digitalocean/datasource_digitalocean_database_cluster.go

```diff
@@ -0,0 +1,140 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceDigitalOceanDatabaseCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDigitalOceanDatabaseClusterRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"engine": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"node_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"maintenance_window": {
+				Type:     schema.TypeList,
+				Computed: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hour": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"host": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"database": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"user": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceDigitalOceanDatabaseClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).godoClient()
+
+	name := d.Get("name").(string)
+
+	opt := &godo.ListOptions{
+		Page:    1,
+		PerPage: 200,
+	}
+
+	databases, _, err := client.Databases.List(context.Background(), opt)
```

Comment:
This should handle pagination in case there are more than 200 Database clusters on an account. Something like:

	var databaseList []godo.Database

	for {
		databases, resp, err := client.Databases.List(context.Background(), opts)
		if err != nil {
			return fmt.Errorf("Error retrieving DatabaseClusters: %s", err)
		}

		for _, d := range databases {
			databaseList = append(databaseList, d)
		}

		if resp.Links == nil || resp.Links.IsLastPage() {
			break
		}

		page, err := resp.Links.CurrentPage()
		if err != nil {
			return fmt.Errorf("Error retrieving DatabaseClusters: %s", err)
		}

		opts.Page = page + 1
	}